src/forwarddiff.jl

```diff
@@ -0,0 +1,9 @@
+using .ForwardDiff: gradient!
+using .DiffResults: GradientResult, value, gradient
```

Comment:
This could be simplified to
```suggestion
using .ForwardDiff: gradient!, DiffResults
using DiffResults: GradientResult, value, gradient
```
(would also allow to simplify the `@require` block). The downside is that it relies on something internal in ForwardDiff, but I guess the dependency on DiffResults is quite stable.

Reply:
Hmm, possibly, but I'll leave as is for now I think